assets/javascripts/kitten/karl/menus/karl-menu.js

```diff
@@ -0,0 +1,148 @@
+import React from 'react'
+import classNames from 'classnames'
+import { KarlMenuHeader } from 'kitten/karl/headers/karl-header'
+
+export class KarlMenu extends React.Component {
+  constructor(props) {
+    super(props)
+
+    this.state = { opened: false }
+    this.handleClick = this.handleClick.bind(this)
+  }
+
+  handleClick() {
+    this.setState({ opened: !this.state.opened })
+  }
+
+  renderOpacity() {
+    if (this.state.opened)
+      return <div className="karl-Opacity" onClick={ this.handleClick } />
+  }
+
+  render() {
+    const karlMenuHeaderClassName = classNames(
+      'karl-Menu',
+      {
+        'is-opened': this.state.opened,
+      },
+    )
+
+    return (
+      <div>
```

Comment:
`A valid React element (or null) must be returned. You may have returned undefined, an array or some other invalid object.` NULL AU CARRÉ @LanF3usT